### ISSUE TITLE
[resolved] SetDomains method does not allow to clear domains

### ISSUE THREAD

**Issue by zeule** [systemd/systemd#9515]
When trying to set the empty domain for an interface, resolved replies "Root domain is not suitable as search domain". 

Please, allow to pass empty domain to `SetDomains()`. At the same time, it seems like (#9472) the empty list is equal to `~.` (and this can be set). If the assumption is correct, please display the empty list as `~.` both in the `resolvectl domain` output and the `Domains` bus property.

`Revert()` does clear the domains list, but clears everything else too, while I need to change only routing.

systemd 239.

**Comment by poettering** [comment]
I am not sure i follow. 

SetDomains() with an empty list will drop all search domains from the interface.

SetDomains() with a single item `~.` means "preferable route lookups to this interface".

The two are very different concepts.

But I am not sure I grok the described problem at all? It is possible to reset the search list, by simply passing the empty list (and not a list with an empty string or so).

**Comment by zeule** [comment]
You are right, DBus call works, but:
``` 
# resolvectl domain eth0    
Link 2 (eth0): ~.
# resolvectl domain eth0 ""
Failed to set domain configuration: Root domain is not suitable as search domain
# resolvectl domain eth0    
Link 2 (eth0): ~.
```

**Comment by poettering** [comment]
> You are right, DBus call works, but:

Ah, hmm, I see what you mean, currently "resolvectl domain" without domain list means "instead of setting, show the current value". Which means the tool has no way to reset the domain search list, as you simply cannot pass an empty list and also say "set the value". This is something we really should fix.

**Comment by zeule** [comment]
Thanks!

**Comment by anshah1216** [comment]
_resolvctl_ is not available in 18.04 so I have to use systemd-resolvd to clear the DNS Domain and I'm having the same issue:
```
$ sudo systemd-resolve -i eth0 --set-domain=""
Failed to set domain configuration: Root domain is not suitable as search domain
```
Is this the wrong syntax? 
Is there any other workaround to clear/remove DNS Domain from an interface?
